src/main/java/com/ita/if103java/ims/service/impl/WarehouseServiceImpl.java

```diff
@@ -194,12 +230,18 @@ public WarehouseDto update(WarehouseDto warehouseDto, UserDetailsImpl user) {
 
     @Override
     public boolean softDelete(Long id, UserDetailsImpl user) {
+        boolean isDelete;
         Warehouse warehouse = warehouseDao.findById(id, user.getUser().getAccountId());
+        if (CollectionUtils.isEmpty(warehouseDao.findChildrenById(warehouse.getParentID(), user.getUser().getAccountId()))) {
+            isDelete = warehouseDao.softDelete(id);
+        } else throw new WarehouseDeleteException("Warehouse have sub warehouses! Firstly you should delete them!");
 
-        boolean isDelete = warehouseDao.softDelete(id);
-        if (isDelete) {
+        if (!savedItemDao.findSavedItemByWarehouseId(warehouse.getId()).isEmpty()) {
+            throw new WarehouseDeleteException("Warehouse is not empty! Firstly you should remove or transfer all items from that warehouse to another");
+        } else if (isDelete) {
             createEvent(user, warehouse, EventName.WAREHOUSE_REMOVED);
         }
+
         return isDelete;
```

Comment:
1. Fix formatting
2. Better to wrote as following in order to avoid else conditions and spaghetti code:
```
@Override
public boolean softDelete(Long id, UserDetailsImpl user) {
    Warehouse warehouse = warehouseDao.findById(id, user.getUser().getAccountId());
    if (CollectionUtils.isNotEmpty(warehouseDao.findChildrenById(warehouse.getParentID(), user.getUser().getAccountId()))) {
        throw new WarehouseDeleteException("Warehouse have sub warehouses! Firstly you should delete them!");
    }
    if (!savedItemDao.findSavedItemByWarehouseId(warehouse.getId()).isEmpty()) {
            throw new WarehouseDeleteException("Warehouse is not empty! Firstly you should remove or transfer all items from that warehouse to another");
        }
    boolean isDelete = warehouseDao.softDelete(id);
    if (isDelete) {
        createEvent(user, warehouse, EventName.WAREHOUSE_REMOVED);
    }
    return isDelete;
}
```